### ISSUE TITLE
error: throw will always call terminate()

### ISSUE THREAD

**Issue by sineang01** [alembic/alembic#227]
Compiling using gcc 6.3.1 under Linux CentOS Linux release 7.6.1810 (Core)

```
FAILED: lib/Alembic/CMakeFiles/Alembic.dir/AbcCoreHDF5/AwImpl.cpp.o 
/opt/rh/devtoolset-6/root/usr/bin/c++  -DDEBUG=1 -DPLATFORM=LINUX -DPLATFORM_LINUX -I/workspace/builder/workspace/CreatePackage/0/.conan/data/OpenEXR/2.3.0_fn.1/foundry/stable/package/97d787b4865ff69a631643964910ab54d318096e/include/OpenEXR -I/workspace/builder/workspace/CreatePackage/0/.conan/data/HDF/5.1.8.7_fn.1/foundry/stable/package/8f98fff07b3b11008d6df44504ae09ea3963d122/include -Ialembic-1.7.10/lib -Ilib -Wno-literal-suffix -std=c++11 -fvisibility=hidden -DH5_USE_18_API -g   -UNDEBUG -Wall -Werror -Wextra -Wno-unused-parameter -fPIC -MD -MT lib/Alembic/CMakeFiles/Alembic.dir/AbcCoreHDF5/AwImpl.cpp.o -MF lib/Alembic/CMakeFiles/Alembic.dir/AbcCoreHDF5/AwImpl.cpp.o.d -o lib/Alembic/CMakeFiles/Alembic.dir/AbcCoreHDF5/AwImpl.cpp.o -c alembic-1.7.10/lib/Alembic/AbcCoreHDF5/AwImpl.cpp
In file included from alembic-1.7.10/lib/Alembic/Util/PlainOldDataType.h:41:0,
                 from alembic-1.7.10/lib/Alembic/Util/Digest.h:41,
                 from alembic-1.7.10/lib/Alembic/Util/All.h:41,
                 from alembic-1.7.10/lib/Alembic/AbcCoreAbstract/Foundation.h:40,
                 from alembic-1.7.10/lib/Alembic/AbcCoreAbstract/ArchiveReader.h:41,
                 from alembic-1.7.10/lib/Alembic/AbcCoreAbstract/All.h:40,
                 from alembic-1.7.10/lib/Alembic/AbcCoreHDF5/Foundation.h:40,
                 from alembic-1.7.10/lib/Alembic/AbcCoreHDF5/AwImpl.h:40,
                 from alembic-1.7.10/lib/Alembic/AbcCoreHDF5/AwImpl.cpp:37:

alembic-1.7.10/lib/Alembic/AbcCoreHDF5/AwImpl.cpp: In destructor ‘virtual Alembic::AbcCoreHDF5::v11::AwImpl::~AwImpl()’:

alembic-1.7.10/lib/Alembic/Util/Exception.h:93:16: error: throw will always call terminate() [-Werror=terminate]
     throw( exc );                                     \
                ^

alembic-1.7.10/lib/Alembic/AbcCoreAbstract/Foundation.h:96:28: note: in expansion of macro ‘ABC_THROW’
 #define ABCA_THROW( TEXT ) ABC_THROW( TEXT )
                            ^~~~~~~~~

alembic-1.7.10/lib/Alembic/AbcCoreHDF5/AwImpl.cpp:261:13: note: in expansion of macro ‘ABCA_THROW’
             ABCA_THROW( strm.str() );
             ^~~~~~~~~~

alembic-1.7.10/lib/Alembic/Util/Exception.h:93:16: note: in C++11 destructors default to noexcept
     throw( exc );                                     \
                ^
alembic-1.7.10/lib/Alembic/AbcCoreAbstract/Foundation.h:96:28: note: in expansion of macro ‘ABC_THROW’
 #define ABCA_THROW( TEXT ) ABC_THROW( TEXT )
                            ^~~~~~~~~

alembic-1.7.10/lib/Alembic/AbcCoreHDF5/AwImpl.cpp:261:13: note: in expansion of macro ‘ABCA_THROW’
             ABCA_THROW( strm.str() );
             ^~~~~~~~~~
```

I believe the error happens because there is a call to ABCA_THROW in the destructor of AwImpl.
Destructors should not throw and by default destructors are noexcept in C++11

(There are more files in AbcCoreHDF5/ directory with the same issue)

**Comment by lamiller0** [comment]
I've been able to reproduce this, thanks for reporting!

**Comment by lamiller0** [comment]
Addressed here:
https://github.com/alembic/alembic/pull/237

**Comment by lamiller0** [comment]
Released in 1.7.12
